06.bowling_object/lib/shot.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Shot
+  MAX_SCORE = 10
+
+  attr_reader :mark
+
+  def initialize(mark)
+    validate_mark(mark)
+
+    @mark = mark.to_s
+  end
+
+  def score
+    @mark == 'X' ? MAX_SCORE : @mark.to_i
+  end
+
+  def self.shot_factory(marks)
+    shots = []
+    marks.each do |mark|
+      shot = Shot.new(mark)
+      shots << shot
+      shots << Shot.new('0') if shot.mark == 'X' && shots.size < 18
```

Comment:
```suggestion
      shots << Shot.new('0') if shot.strike? && shots.size < 18
```

データを隠蔽して、メソッドを提供するだけにした方がオブジェクト指向らしいです。
（その方がいざというときに内部的なデータ構造を変更しやすい）

あと18はマジックナンバーかな、と思います